test/test_irods_http_api.py

```diff
@@ -3110,6 +3110,263 @@ def test_server_does_not_segfault_when_parallel_write_init_is_invoked_with_insuf
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            result = r.json()
+            self.assertEqual(result['irods_response']['status_code'], 0)
+            self.assertEqual(result['rows'][0][1], str(data_size_0))
+            self.assertEqual(result['rows'][1][1], str(data_size_1))
+
+        finally:
+            # Remove the data object.
+            r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+                'op': 'remove',
+                'lpath': data_object,
+                'catalog-only': 0,
+                'no-trash': 1
+            })
+            self.logger.debug(r.content)
+
+            # Remove resource.
+            r = requests.post(f'{self.url_base}/resources', headers=rodsadmin_headers, data={
+                'op': 'remove',
+                'name': resc_name
+            })
+            self.logger.debug(r.content)
+
+    def test_truncating_a_nonexistent_data_object_returns_an_error(self):
+        headers = {'Authorization': 'Bearer ' + self.rodsuser_bearer_token}
+        data_object = f'/{self.zone_name}/home/{self.rodsuser_username}/truncating_nonexistent_replica.txt'
+
+        r = requests.post(self.url_endpoint, headers=headers, data={
+            'op': 'truncate',
+            'lpath': data_object,
+            'size': 1024
+        })
+        self.logger.debug(r.content)
+        self.assertEqual(r.status_code, 200)
+        self.assertEqual(r.json()['irods_response']['status_code'], irods_error_codes.OBJ_PATH_DOES_NOT_EXIST)
+
+    def test_truncating_a_replica_using_incompatible_parameters(self):
+        rodsuser_headers = {'Authorization': f'Bearer {self.rodsuser_bearer_token}'}
+        data_object = f'/{self.zone_name}/home/{self.rodsuser_username}/truncating_using_incompatible_params.txt'
+
+        try:
+            # Create a data object.
+            r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+                'op': 'touch',
+                'lpath': data_object
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+            # Show that the truncate operation returns an error when a replica number
+            # and [root] resource is specified.
+            r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+                'op': 'truncate',
+                'lpath': data_object,
+                'size': 100,
+                'replica-number': 0,
+                'resource': 'demoResc'
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.json()['irods_response']['status_code'], irods_error_codes.SYS_INVALID_INPUT_PARAM)
+
+            # TODO(???): Enable once the truncate API honors this case.
```

Comment:
Yep, we should probably make sure this doesn't happen... I don't think any of our APIs enforce this.